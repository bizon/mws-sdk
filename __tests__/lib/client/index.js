const MWSClient = require('../../../lib/client')

describe('lib.client.index', () => {
  it('should fail if accessKeyId, secretAccessKey, sellerId or mwsToken is not specified', () => {
    const tests = [
      () => new MWSClient(),
      () => new MWSClient({
        accessKeyId: 'foo'
      }),
      () => new MWSClient({
        accessKeyId: 'foo',
        secretAccessKey: 'bar'
      }),
      () => new MWSClient({
        accessKeyId: 'foo',
        secretAccessKey: 'bar',
        sellerId: 'baz'
      })
    ]

    for (const test of tests) {
      expect(test).toThrow('accessKeyId, secretAccessKey, sellerId and mwsToken are required')
    }
  })

  it('should fail if the region is unknown', () => {
    expect(
      () => new MWSClient({
        accessKeyId: 'foo',
        secretAccessKey: 'bar',
        sellerId: 'baz',
        mwsToken: 'token',
        sellerRegion: 'unknown'
      })
    ).toThrow('Unknown region unknown')
  })

  it('should grab accessKeyId and secretAccessKey from the environment', () => {
    process.env.MWS_ACCESS_KEY_ID = 'foo'
    process.env.MWS_SECRET_ACCESS_KEY = 'bar'

    const client = new MWSClient({
      sellerId: 'baz',
      mwsToken: 'token',
      sellerRegion: 'eu'
    })

    expect(client.settings.accessKeyId).toBe(process.env.MWS_ACCESS_KEY_ID)
    expect(client.settings.secretAccessKey).toBe(process.env.MWS_SECRET_ACCESS_KEY)

    delete process.env.MWS_ACCESS_KEY_ID
    delete process.env.MWS_SECRET_ACCESS_KEY
  })
})
