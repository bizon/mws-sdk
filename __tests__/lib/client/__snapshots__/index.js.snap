// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.index should not enumerate secret properties 1`] = `
Array [
  "sellerId",
  "mwsDomain",
  "marketplaces",
]
`;

exports[`lib.client.index should throw default a MWSError when encountering an unhandled HTTPError on GET 1`] = `
Object {
  "error": Object {
    "code": "QuotaExceeded",
    "detail": null,
    "message": "You exceeded your quota of 200.0 requests per 1 hour for operation CustomResource/1988-10-13/CustomAction.  Your quota will reset on Thu Apr 04 18:46:00 UTC 2020",
    "type": "",
  },
  "responseMetadata": Object {
    "requestId": "bc6e4601-3d74-4612-adcf-EXAMPLEf1796",
  },
}
`;

exports[`lib.client.index should throw default a MWSError when encountering an unhandled HTTPError on POST 1`] = `
Object {
  "error": Object {
    "code": "ClientError",
    "detail": null,
    "message": "bar is not valid for Foo",
    "type": null,
  },
  "responseMetadata": Object {
    "requestId": "bc6e4601-3d74-4612-adcf-EXAMPLEf1796",
  },
}
`;
